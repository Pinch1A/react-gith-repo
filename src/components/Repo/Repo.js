import React from 'react'

import styled from 'styled-components'

const Row = styled.div`
  padding: 5px 0;
`
const Column = styled.div`
  display: flex;
  padding: 2px;
  justify-content: space-between;
`
const Cell = styled.div`
  display: flex;
  padding: 2px;
`

const Stars = styled(Cell)`
  color: red;
  font-weight: 800;
`

const StyledDesc = styled(Cell)`
  font-size: 14px;
  color: grey;
  max-width: 400px;
`

const Repo = ({
  repository: {
    name,
    description,
    stargazers: { totalCount },
    url,
  },
}) => {
  return (
    <Row>
      <Column>
        <a href={url} target="_blank" rel="noopener noreferrer">
          <Cell>{name}</Cell>
        </a>
        <Stars>{totalCount}</Stars>
      </Column>
      <div>
        <StyledDesc>{description}</StyledDesc>
      </div>
    </Row>
  )
}

export default Repo
