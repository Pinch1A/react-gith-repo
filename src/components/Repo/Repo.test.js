import React from 'react'
import Repo from './Repo'
import { shallow } from 'enzyme'

const repository = {
  name: 'test',
  description: 'mock test',
  stargazers: { totalCount: 100 },
  url: 'test.test',
}
describe('<Repo>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Repo repository={repository} />)
    expect(wrapper).toMatchSnapshot()
  })
})
