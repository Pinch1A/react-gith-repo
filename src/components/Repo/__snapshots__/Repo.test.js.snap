// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repo> renders without crashing 1`] = `
<styled.div>
  <styled.div>
    <a
      href="test.test"
      rel="noopener noreferrer"
      target="_blank"
    >
      <styled.div>
        test
      </styled.div>
    </a>
    <Styled(styled.div)>
      100
    </Styled(styled.div)>
  </styled.div>
  <div>
    <Styled(styled.div)>
      mock test
    </Styled(styled.div)>
  </div>
</styled.div>
`;
